sdk/translation/azure-ai-translation-text/src/main/java/com/azure/ai/translation/text/TextTranslationClientBuilder.java

```diff
@@ -266,14 +272,27 @@ public TextTranslationClientBuilder credential(KeyCredential credential) {
      *
      * @param region where the Translator resource is created.
      * @return The updated {@link TextTranslationClientBuilder} object.
-     * @throws NullPointerException If {@code tokenCredential} is null.
+     * @throws NullPointerException If {@code region} is null.
      */
     public TextTranslationClientBuilder region(String region) {
         Objects.requireNonNull(region, "'region' cannot be null.");
         this.region = region;
         return this;
     }
 
+    /**
+     * Sets the Azure Resource Id used to authorize requests sent to the service.
+     *
+     * @param azureResourceId Id of the Translator Resource.
+     * @return The updated {@link TextTranslationClientBuilder} object.
+     * @throws NullPointerException If {@code azureResourceId} is null.
+     */
+    public TextTranslationClientBuilder azureResourceId(String azureResourceId) {
+        Objects.requireNonNull(azureResourceId, "'azureResourceId' cannot be null.");
+        this.azureResourceId = azureResourceId;
```

Comment:
If this is set, will the client be able to use other resources apart from the one set here?

Reply:
No, user will need to create another client with different resource id - it is part of authetication. https://learn.microsoft.com/en-us/azure/ai-services/translator/reference/v3-0-reference#authentication-with-microsoft-entra-id